parquet-column/src/main/java/org/apache/parquet/column/MinMax.java

```diff
@@ -47,43 +39,22 @@ public T getMax() {
     return max;
   }
 
-  private void getMinAndMax() {
-    while(iterator.hasNext())  {
-      T element = iterator.next();
+  private void getMinAndMax(PrimitiveComparator comparator, Iterable<T> iterable) {
+    iterable.forEach(element -> {
       if (max == null) {
         max = element;
-      } else if (max != null && element != null) {
-        if ((element instanceof Integer &&
-          ((PrimitiveComparator<Integer>)comparator).compare((Integer)max, (Integer)element) < 0) ||
-          (element instanceof Binary &&
-            ((PrimitiveComparator<Binary>)comparator).compare((Binary)max, (Binary)element) < 0) ||
-          (element instanceof Double &&
-             ((PrimitiveComparator<Double>)comparator).compare((Double)max, (Double)element) < 0) ||
-          (element instanceof Float &&
-             ((PrimitiveComparator<Float>)comparator).compare((Float)max, (Float)element) < 0) ||
-          (element instanceof Boolean &&
-            ((PrimitiveComparator<Boolean>)comparator).compare((Boolean)max, (Boolean)element) < 0) ||
-          (element instanceof Long &&
-            ((PrimitiveComparator<Long>)comparator).compare((Long) max, (Long)element) < 0))
+      } else if (element != null) {
+        if (comparator.compare(max, element) < 0) {
```

Comment:
nit: You may combine the two with an `&&`.

Reply:
Done.